src/test_fixture/mod.rs

```diff
@@ -136,3 +136,13 @@ pub fn permutation_valid(permutation: &[u32]) -> bool {
     }
     true
 }
+
+/// Take a slice of bits in `{0,1} ⊆ F_p`, and reconstruct the integer in `F_p`
+pub fn bits_to_field<F: Field>(x: &[F]) -> F {
+    #[allow(clippy::cast_possible_truncation)]
+    let v = x
+        .iter()
+        .enumerate()
+        .fold(0, |acc, (i, &b)| acc + 2_u128.pow(i as u32) * b.as_u128());
```

Comment:
I think there is something *really* subtle going on here.

Because the {0, 1} values have type of `F: Field` they are *wrapping around* when we add them up here. 

That means that if we gave it an input like:
```
[1, 1, 1, 1, 1, 1, 1, 1]
```
This function is going to do:
1+2+4+8+16+32+64+128 = 255
Fp31::from(255)
# => 7
```

...which is kind of like saying:

"[1, 1, 1, 1, 1, 1, 1, 1] is the bitwise representation of 7"

...and that's not true... If we performed a "bit decomposition" of the number 7 and got [1, 1, 1, 1, 1, 1, 1, 1] we wouldn't be happy, right?

Reply:
I can't help myself:

```suggestion
        .fold(0, |acc, (i, &b)| acc + (b.as_u128() << (i as u32)));
```

On the substantive part: why would get a value of 255 in like that?

Reply:
Oh that is kind of slick =). Will clippy complain about possible truncation or something?

On the substantive part:

We are literally generating random bits. So for each bit there are 50/50 odds of getting a 1 or a 0. So one time in 256 runs, you'll get 0b11111111

Reply:
I thought that that was why the values of the random fields were being clamped to the range of F, rather than taking the raw u128.

Reply:
I created this method to test BitwiseSum, which could return a sequence of bits outside the range of `F`. I created another method `bits_to_value() -> u128` that doesn't wrap the integer into a field.